packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/EnumTypeConverter.kt

```diff
@@ -25,6 +27,12 @@ class EnumTypeConverter(
     "Cannot convert js value to enum without the primary constructor"
   }
 
+  init {
+    if (!enumClass.isSubclassOf(EnumArgument::class)) {
+      logger.warn("Enum '${enumClass}' should inherit from ${EnumArgument::class}.")
+    }
```

Comment:
I didn't want to crash the application here. So I've added a warning here. Not sure if it was the best idea. What do you think?